crates/walrus-service/src/event/event_processor/db.rs

```diff
@@ -149,4 +149,34 @@ impl EventProcessorStores {
         self.walrus_package_store.schedule_delete_all()?;
         Ok(())
     }
+
+    /// Inserts a checkpoint into the checkpoint store in a batch.
+    /// Inv: it should never insert smaller sequence number than the current latest checkpoint
+    /// sequence number.
+    pub fn insert_checkpoint_in_batch(
+        &self,
+        batch: &mut DBBatch,
+        checkpoint: &VerifiedCheckpoint,
+    ) -> Result<(), TypedStoreError> {
+        #[cfg(debug_assertions)]
```

Comment:
it'll be nice to have this invariant check enabled in prod. But we are doing a DB read on every checkpoint, which I tried to avoid.